hnn_core/network.py

```diff
@@ -374,23 +381,31 @@ def add_evoked_drive(self, name, *, mu, sigma, numspikes,
         if not self._legacy_mode:
             _check_drive_parameter_values('evoked', sigma=sigma,
                                           numspikes=numspikes)
-
         drive = _NetworkDrive()
         drive['type'] = 'evoked'
         if name == 'extgauss':
             drive['type'] = 'gaussian'  # XXX needed to pass legacy tests!
-        drive['cell_specific'] = True
+        if n_drive_cells == 'n_cells':
+            n_drive_cells = self.n_cells
+            drive['cell_specific'] = True
+        else:
+            _validate_type(n_drive_cells, types=int)
+            _check_option('n_drive_cells', n_drive_cells, allowed_values=[1])
```

Comment:
I can't quite wrap my head around this yet. Why say `int` in the docstring if only one value is allowed? Shouldn't `n_drive_cells` work consistently for all the drives?

Reply:
I guess it's not obvious to me either why you are only allowed 1 artificial cell for an evoked drive.

Reply:
Oh my bad, we can change this. I put this in at the very beginning while I was still trying to figure out how much flexibility to build into the API.

Reply:
Thinking about this more, I suppose the usefulness of multiple independent cells isn't really shown when they have all-to-all connectivity. 

However, it becomes more important when you consider #369 as that can introduce the possibility of each cell potentially targeting a different subset of targets. Possibly through: https://github.com/jonescompneurolab/hnn-core/blob/08daf2024467b46e4efc447efd935645b4fd0aca/hnn_core/network.py#L96

Reply:
I guess this is baking in "repeats" in the "evoked" drive which historically didn't have this parameter. I'm fine with this if it's consistent with the "rhythmic" drive. We will need a follow up PR though @rythorpe . It's clear there is more work needed here.

Reply:
Correct, this PR is more about eliminating/consolidating confusing concepts in the API like `repeats` for the rhythmic drives and `sync_within_trials` for the evoked drives than it is about building in novel functionality. As I recently discovered in https://github.com/jonescompneurolab/hnn-core/pull/383/commits/897bbc6612f1e9534a4e311aee28e8dfc268460d, there are also some odd conventions that might produce silent bugs later on if not addressed.